module/os/linux/zfs/arc_os.c

```diff
@@ -464,13 +462,31 @@ arc_memory_throttle(spa_t *spa, uint64_t reserve, uint64_t txg)
 uint64_t
 arc_all_memory(void)
 {
-	return (ptob(physmem) / 2);
+	return (ptob(physmem));
+}
+
+void
+arc_shrink(size_t percent)
+{
+	if (percent > 80) {
+		percent = 80;
+	}
+
+	uint64_t to_free = aggsum_value(&arc_sums.arcstat_size) * percent / 100;
+	arc_reduce_target_size(to_free);
+	arc_wait_for_eviction(to_free, B_FALSE);
+	arc_no_grow = B_TRUE;
 }
 
 uint64_t
 arc_free_memory(void)
 {
-	return (random_in_range(arc_all_memory() * 20 / 100));
+	struct sysinfo info;
+	if (sysinfo(&info) == -1) {
```

Comment:
Will this function be invoked in regular IO path? if yes, we need to measure the overhead of this syscall.

Reply:
1. Calls to this function are not in critical path
2. This function cost only 0.5us~1us